### PR TITLE
✨Search group menu

### PR DESCRIPTION
**Pull request by AntoLC**
## Purpose

Integrate the design and partial interactions of the search group menu.

## Proposal

- [x] Integrate the design
- [x] Display teams depend the cases (no teams / teams empty / teams with users)
- [x] Add actions interactions 
- [x] Add Infinite scroll bar
- [x] more tests

## Demo

[scrnli_2_2_2024_4-01-37 PM.webm](https://github.com/numerique-gouv/people/assets/25994652/0c0a6242-8137-431a-abf7-0c59c4067550)

### CODE REVIEW

File: src/backend/core/api/viewsets.py [hunk 1]
```diff
@@ -244,6 +245,9 @@ class TeamViewSet(
 
     permission_classes = [permissions.AccessPermission]
     serializer_class = serializers.TeamSerializer
+    filter_backends = [filters.OrderingFilter]
+    ordering_fields = ["created_at"]
```
Comment (mjeammet) [inline src/backend/core/api/viewsets.py:249]: Nice one! Do you think we could also provide the option of ordering by name ?

**Review by mjeammet — APPROVED** [review]
See my two small comments but otherwise it looks good to me !

File: src/backend/core/api/viewsets.py (hunk 1 above)
Comment (AntoLC) [reply src/backend/core/api/viewsets.py:249]: Yes you can (https://www.django-rest-framework.org/api-guide/filtering/#specifying-which-fields-may-be-ordered-against), but I don't need it in this PR, let's add it when we will need it.

File: src/backend/core/tests/test_api_teams.py
[the remark this answers is not in the document]
Comment (AntoLC) [reply src/backend/core/tests/test_api_teams.py:1]: [fixup! ♻️(backend) api teams list ordering](https://github.com/numerique-gouv/people/pull/31/commits/2ce7c2adeb68be40c91747e78822707dfd623b7f)

File: src/frontend/apps/desk/src/features/teams/store/useTeamsStore.tsx [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+export const useTeamStore = create<TeamsStore>((set) => ({
+  ordering: TeamsOrdering.BY_CREATED_ON_DESC,
+  changeOrdering: () =>
+    set(({ ordering }) => ({
+      ordering:
+        ordering === TeamsOrdering.BY_CREATED_ON
+          ? TeamsOrdering.BY_CREATED_ON_DESC
+          : TeamsOrdering.BY_CREATED_ON,
+    })),
+}));
```
Comment (lebaudantoine) [inline src/frontend/apps/desk/src/features/teams/store/useTeamsStore.tsx:19]: Would it be easier to store a boolean flag `is_descending` or smth? Do you plan to introduce new `TeamsOrdering` values?

Comment (lebaudantoine) [reply src/frontend/apps/desk/src/features/teams/store/useTeamsStore.tsx:19]: Let's check with the product team, what they add in mind, if it was date ordering, or name ordering btw.

File: src/frontend/apps/desk/src/features/teams/components/PanelTeams.tsx [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+type DictTeams = {
+  [page: string]: TeamResponse[];
+};
+
+export const PanelTeams = () => {
+  const { t } = useTranslation();
+
+  const [dictTeams, setDictTeams] = useState<DictTeams>({});
+  const [page, setPage] = useState(1);
+  const ordering = useTeamStore((state) => state.ordering);
```
Comment (lebaudantoine) [inline src/frontend/apps/desk/src/features/teams/components/PanelTeams.tsx:22]: smth as `const {ordering} = useTemaStore()`would not work? Sorry I am not a zustand expert

File: src/frontend/apps/desk/src/features/teams/api/useCreateTeam.tsx [hunk 4]
```diff
@@ -34,7 +34,6 @@ export function useCreateTeam() {
     onSuccess: () => {
       void queryClient.invalidateQueries({
         queryKey: [KEY_LIST_TEAM],
-        exact: true,
```
Comment (lebaudantoine) [inline src/frontend/apps/desk/src/features/teams/api/useCreateTeam.tsx:37]: Sorry I am a bit lost, why did you remove this?

File: src/frontend/apps/desk/src/features/teams/api/useTeams.tsx [hunk 5]
```diff
@@ -2,18 +2,44 @@ import { UseQueryOptions, useQuery } from '@tanstack/react-query';
...
 type TeamsResponse = APIList<TeamResponse>;
 export interface TeamsResponseError {
   detail: string;
 }
 
-export const getTeams = async () => {
-  const response = await fetchAPI(`teams/`);
+export const getTeams = async ({ ordering, page }: TeamsParams) => {
+  const orderingQuery = ordering ? `&ordering=${ordering}` : '';
+  const response = await fetchAPI(`teams/?page=${page}${orderingQuery}`);
```
Comment (lebaudantoine) [inline src/frontend/apps/desk/src/features/teams/api/useTeams.tsx:42]: is there any "nicer" way to join/stringify url params with fetch? This workaround looks error prone

File: src/frontend/apps/desk/src/features/teams/api/useTeams.tsx [hunk 6]
```diff
@@ -2,18 +2,44 @@ import { UseQueryOptions, useQuery } from '@tanstack/react-query';
...
+  role: string;
+  user: string;
+  abilities: {
+    get: boolean;
+    patch: boolean;
+    put: boolean;
+    delete: boolean;
+    manage_accesses: boolean;
+    set_role_to: [];
+  };
```
Comment (lebaudantoine) [inline src/frontend/apps/desk/src/features/teams/api/useTeams.tsx:16]: do we need `abilities`? User's role should be enough

File: src/frontend/apps/desk/src/features/teams/api/useTeams.tsx [hunk 7]
```diff
@@ -2,18 +2,44 @@ import { UseQueryOptions, useQuery } from '@tanstack/react-query';
 
 import { APIList, fetchAPI } from '@/api';
 
-interface TeamResponse {
+interface Access {
+  id: string;
+  role: string;
```
Comment (lebaudantoine) [inline src/frontend/apps/desk/src/features/teams/api/useTeams.tsx:7]: should be an Enum

File: src/frontend/apps/desk/src/features/teams/components/PanelActions.tsx [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+      background-color: ${({ $background }) => $background};
+      border-radius: 10rem;
+      color: ${({ $color }) => $color};
+    }
+  }
+`;
+
+export const PanelActions = () => {
+  const { t } = useTranslation();
+  const { changeOrdering, ordering } = useTeamStore((state) => state);
```
Comment (lebaudantoine) [inline src/frontend/apps/desk/src/features/teams/components/PanelActions.tsx:34]: ditto, again I am a noob with zustand, is smht as `const { changeOrdering, ordering } = useTeamStore();`would work?

File: docker/auth/realm.json [hunk 9]
```diff
@@ -56,12 +56,34 @@
...
+      ],
+      "realmRoles": ["user"]
+    },
+    {
+      "username": "user-e2e-firefox",
+      "enabled": true,
+      "credentials": [
+        {
+          "type": "password",
+          "value": "password-e2e-firefox"
```
Comment (lebaudantoine) [inline docker/auth/realm.json:86]: Where is the logic binding the username with the tested browser?

File: src/frontend/apps/e2e/__tests__/app-desk/menu.spec.ts [hunk 10]
```diff
@@ -2,9 +2,9 @@ import { expect, test } from '@playwright/test';
 
 import { keyCloakSignIn } from './common';
 
-test.beforeEach(async ({ page }) => {
+test.beforeEach(async ({ page, browserName }) => {
   await page.goto('/');
-  await keyCloakSignIn(page);
+  await keyCloakSignIn(page, browserName);
```
Comment (lebaudantoine) [inline src/frontend/apps/e2e/__tests__/app-desk/menu.spec.ts:7]: it quite confusing, to add a `browserName` param to the `keycloakSignIn` function

File: src/frontend/packages/i18n/__tests__/translations.test.ts [hunk 11]
```diff
@@ -20,9 +20,29 @@ describe('checks all the frontend translation are made', () => {
...
+            console.log(
+              `Missing keys in Desk translations that should be translated in Crowdin, got to https://crowdin.com/:`,
+              missingKeys,
+            );
+          }
+          if (additionalKeys.length > 0) {
+            console.log(
+              `Additional keys in Desk translations that seems not present in this branch:`,
+              additionalKeys,
+            );
```
Comment (lebaudantoine) [inline src/frontend/packages/i18n/__tests__/translations.test.ts:41]: It feels weird to console.log such message, but I have no better suggestion yet. Is it the right way to log some error while running the tests?

File: src/frontend/apps/desk/src/features/teams/api/useTeams.tsx [hunk 12]
```diff
@@ -24,15 +50,16 @@ export const getTeams = async () => {
...
 
 export function useTeams(
+  param: TeamsParams,
   queryConfig?: UseQueryOptions<
     TeamsResponse,
     TeamsResponseError,
     TeamsResponse
   >,
 ) {
   return useQuery<TeamsResponse, TeamsResponseError, TeamsResponse>({
```
Comment (lebaudantoine) [inline src/frontend/apps/desk/src/features/teams/api/useTeams.tsx:60]: Should we use [useInfiniteQuery](https://tanstack.com/query/latest/docs/framework/react/reference/useInfiniteQuery) instead?

File: src/frontend/apps/desk/src/components/InfiniteScroll.tsx [hunk 13]
```diff
@@ -0,0 +1,55 @@
...
+      }
+
+      // To not wait until the end of the scroll to load more data
+      const loadBefore = 150;
+
+      const { scrollTop, clientHeight, scrollHeight } = scrollContainer;
+      if (scrollTop + clientHeight >= scrollHeight - loadBefore) {
+        next();
+      }
+    };
```
Comment (lebaudantoine) [inline src/frontend/apps/desk/src/components/InfiniteScroll.tsx:39]: if we use `useInfiniteQuery` this would be refactored right?

File: src/frontend/apps/desk/src/components/InfiniteScroll.tsx [hunk 14]
```diff
@@ -0,0 +1,55 @@
...
+
+    const handleScroll = () => {
+      if (timeout.current) {
+        clearTimeout(timeout.current);
+      }
+      timeout.current = setTimeout(nextHandle, 50);
```
Comment (lebaudantoine) [inline src/frontend/apps/desk/src/components/InfiniteScroll.tsx:45]: why do you use a timeout on this one?

File: src/frontend/apps/desk/src/features/teams/components/PanelTeams.tsx [hunk 15]
```diff
@@ -0,0 +1,129 @@
...
+        ...newTeams,
+      };
+    });
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [data?.results]);
+
+  return (
+    <Box $height="90vh" $css="overflow: auto;" ref={containerRef}>
+      <InfiniteScroll
+        hasMore={data?.count ? data.count > teams.length : false}
```
Comment (lebaudantoine) [inline src/frontend/apps/desk/src/features/teams/components/PanelTeams.tsx:76]: nit: should it work?
```suggestion
        hasMore={data?.count > teams.length}
```

File: src/frontend/apps/desk/src/features/teams/components/PanelTeams.tsx [hunk 16]
```diff
@@ -0,0 +1,129 @@
...
+            </Text>
+          </Box>
+        )}
+
+        {!isError &&
+          teams.map((team) => (
+            <Fragment key={team.id}>
+              <PanelTeam team={team} />
+            </Fragment>
+          ))}
```
Comment (lebaudantoine) [inline src/frontend/apps/desk/src/features/teams/components/PanelTeams.tsx:125]: As, `isLoading`, `data && !data.count` and `isError` are not exclusive, these 3 blocks could be displayed altogether?

File: src/frontend/apps/desk/src/features/teams/components/PanelTeam.tsx [hunk 17]
```diff
@@ -0,0 +1,47 @@
+import React from 'react';
```
Comment (lebaudantoine) [inline src/frontend/apps/desk/src/features/teams/components/PanelTeam.tsx:1]: I would change this naming, as it's more a List item, and `PanelTeams` a list

File: src/frontend/apps/desk/src/features/teams/components/PanelTeams.tsx [hunk 18]
```diff
@@ -0,0 +1,129 @@
...
+    page,
+    ordering,
+  });
+  const containerRef = useRef<HTMLDivElement>(null);
+  const scrollLoading = useRef(false);
+  const teams = useMemo(() => {
+    let teams: TeamResponse[] = [];
+    Object.keys(dictTeams)
+      .sort()
+      .forEach((key) => (teams = teams.concat(dictTeams[key])));
```
Comment (lebaudantoine) [inline src/frontend/apps/desk/src/features/teams/components/PanelTeams.tsx:33]: I think this logic would be refactored with the infinite query hook, but let's discuss it orally

**Review by lebaudantoine — COMMENTED** [review]
Thanks for the MASSIIIIIIVE PR! Haven't had a chance to check out the Team test yet, but it's awesome to see the screen come together. Exciting review ahead! 👷

File: src/frontend/apps/desk/src/app/globals.css [hunk 19]
```diff
@@ -4,3 +4,22 @@ body {
...
+::-webkit-scrollbar-thumb {
+  background-color: #d6dee1;
+  border-radius: 20px;
+  border: 6px solid transparent;
+  background-clip: content-box;
+}
+
+::-webkit-scrollbar-thumb:hover {
+  background-color: #a8bbbf;
+}
```
[the remark this answers is not in the document]
Comment (manuhabitela) [reply src/frontend/apps/desk/src/app/globals.css:25]: But are the width and border-radius actually intended to be the same at all times?

[the remark this answers is not in the document]
Comment (lebaudantoine) [reply src/frontend/apps/desk/src/app/globals.css:25]: It's a guess, but I am pretty these two quantities match to have a nice rounded thumb in the scroll bar. Actually, I would suggest replacing the `border-radius: 20px;`  by a large pixel size `border-radius: 9999px;` (inspiration from tailwind rounded-full class).

File: src/frontend/apps/desk/src/i18n/translations.json [hunk 20]
```diff
@@ -1,30 +1,40 @@
...
       "Groups": "Groupes",
-      "Help": "Aide",
-      "Profile picture": "Photo de l'utilisateur",
       "{{label}} button": "Bouton {{label}}",
       "{{label}} icon": "Icône {{label}}",
-      "Team name": "Nom de l’équipe",
-      "Create Team": "Créer une équipe"
+      "Recents": "Récents",
+      "Sort the teams": "Trier les équipes",
+      "Sort teams icon": "Icône trier les groupes",
```
[the remark this answers is not in the document]
Comment (lebaudantoine) [reply src/frontend/apps/desk/src/i18n/translations.json:28]: Big shoutout to @manuhabitela! 🌟 Your response is packed with so much info. 😱🙏 Thanks for taking the time to share your knowledge! 🚀

File: src/frontend/packages/i18n/__tests__/translations.test.ts [hunk 21]
```diff
@@ -20,9 +20,29 @@ describe('checks all the frontend translation are made', () => {
...
-          listKeysCrowdin.every((element) => listKeysDesk.includes(element)),
-        ).toBeTruthy();
+        const missingKeys = listKeysCrowdin.filter(
+          (element) => !listKeysDesk.includes(element),
+        );
+        const additionalKeys = listKeysDesk.filter(
+          (element) => !listKeysCrowdin.includes(element),
+        );
+
+        if (missingKeys.length > 0 || additionalKeys.length > 0) {
```
[the remark this answers is not in the document]
Comment (AntoLC) [reply src/frontend/packages/i18n/__tests__/translations.test.ts]: https://github.com/numerique-gouv/people/pull/31/commits/26597a338b0710a8f114435becca6f143d7273de

File: src/frontend/apps/desk/src/components/Box.tsx [hunk 22]
```diff
@@ -19,8 +19,11 @@ export interface BoxProps {
   $width?: CSSProperties['width'];
 }
 
+export type BoxType = ComponentPropsWithRef<typeof Box>;
+
 export const Box = styled('div')<BoxProps>`
   display: flex;
+  flex-direction: column;
```
[the remark this answers is not in the document]
Comment (AntoLC) [reply src/frontend/apps/desk/src/components/Box.tsx:26]: https://github.com/numerique-gouv/people/pull/31/commits/9ffe6ce2388750bd64aaa8c68a66f3b2634be99f

File: src/frontend/apps/desk/src/app/page.tsx [hunk 23]
```diff
@@ -1,17 +1,33 @@
...
   const { t } = useTranslation();
 
   return (
-    <Box $direction="column" className="p-b">
-      <h1>{t('Hello Desk !')}</h1>
-      <Teams />
+    <Box $height="inherit" $direction="row">
+      <Panel />
+      <Box
+        $background="#FAFAFA"
```
[the remark this answers is not in the document]
Comment (AntoLC) [reply src/frontend/apps/desk/src/app/page.tsx]: https://github.com/numerique-gouv/people/pull/31/commits/eb7af0f41bfd043adf1a6ae180597e6a50e1ae6f

File: src/frontend/apps/desk/src/features/teams/index.tsx [hunk 24]
```diff
@@ -1,49 +1,2 @@
...
-                </span>
-              </div>
-            </li>
-          ))}
-        </ul>
-      </section>
-    </>
-  );
-};
+export * from './components/Panel';
```
[the remark this answers is not in the document]
Comment (AntoLC) [reply src/frontend/apps/desk/src/features/teams/index.tsx]: https://github.com/numerique-gouv/people/pull/31/commits/669aa39320299052bfcb86f2201f2b2bc2c32671#diff-e29a68b2b59d77584a116ebfa54f2f20fd85f7a95359f681b5a3fb7b4fab2560

File: src/frontend/apps/desk/src/i18n/translations.json [hunk 25]
```diff
@@ -1,30 +1,40 @@
...
+      "Add a team": "Ajouter une équipe",
+      "Add team icon": "Icône ajout d'une équipe",
+      "Teams icon": "Icône de groupe",
+      "Empty teams icon": "Icône de groupe vide",
+      "0 group to display": "0 groupe à afficher.",
```
[the remark this answers is not in the document]
Comment (AntoLC) [reply src/frontend/apps/desk/src/i18n/translations.json]: https://github.com/numerique-gouv/people/pull/31/commits/26597a338b0710a8f114435becca6f143d7273de

File: src/backend/core/tests/teams/test_core_api_teams_list.py [hunk 26]
```diff
@@ -116,3 +115,59 @@ def test_api_teams_list_authenticated_distinct():
     content = response.json()
     assert len(content["results"]) == 1
     assert content["results"][0]["id"] == str(team.id)
+
+
+def test_api_teams_order():
+    """
+    Test that the endpoint GET teams is sorted in created_at descending order by default.
```
[the remark this answers is not in the document]
Comment (AntoLC) [reply src/backend/core/tests/teams/test_core_api_teams_list.py]: https://github.com/numerique-gouv/people/pull/31/commits/040b49377984ccccba9f96faa0fad557c912acb8

File: src/frontend/apps/desk/src/i18n/translations.json [hunk 27]
```diff
@@ -1,30 +1,40 @@
 {
-  "en": { "translation": {} },
   "fr": {
     "translation": {
-      "Hello Desk !": "Bonjour Desk !",
+      "Create a new team": "Créer un nouveau groupe",
+      "Team name": "Nom du groupe",
+      "Create Team": "Créer un groupe",
```
[the remark this answers is not in the document]
Comment (AntoLC) [reply src/frontend/apps/desk/src/i18n/translations.json]: https://github.com/numerique-gouv/people/pull/31/commits/669aa39320299052bfcb86f2201f2b2bc2c32671

File: src/frontend/apps/desk/src/features/teams/components/PanelTeams.tsx (hunk 3 above)
Comment (AntoLC) [reply src/frontend/apps/desk/src/features/teams/components/PanelTeams.tsx]: https://github.com/numerique-gouv/people/pull/31/commits/669aa39320299052bfcb86f2201f2b2bc2c32671

File: src/frontend/apps/desk/src/features/teams/api/useTeams.tsx (hunk 7 above)
Comment (AntoLC) [reply src/frontend/apps/desk/src/features/teams/api/useTeams.tsx]: https://github.com/numerique-gouv/people/pull/31/commits/31c9e6cfa3e116c7ff64a68c31e4339de408c3d6

File: src/frontend/apps/desk/src/features/teams/components/PanelActions.tsx (hunk 8 above)
Comment (AntoLC) [reply src/frontend/apps/desk/src/features/teams/components/PanelActions.tsx]: https://github.com/numerique-gouv/people/pull/31/commits/eb7af0f41bfd043adf1a6ae180597e6a50e1ae6f

File: src/frontend/apps/desk/src/features/teams/api/useTeams.tsx (hunk 12 above)
Comment (AntoLC) [reply src/frontend/apps/desk/src/features/teams/api/useTeams.tsx]: 👍  It gives us a good refacto
https://github.com/numerique-gouv/people/pull/31/commits/ccef7d82b26ff1a60c90314be58c89d48aa38580

File: src/frontend/apps/desk/src/features/teams/components/PanelTeams.tsx (hunk 16 above)
Comment (AntoLC) [reply src/frontend/apps/desk/src/features/teams/components/PanelTeams.tsx]: https://github.com/numerique-gouv/people/pull/31/commits/ccef7d82b26ff1a60c90314be58c89d48aa38580

File: src/frontend/apps/desk/src/features/teams/components/PanelTeams.tsx (hunk 18 above)
Comment (AntoLC) [reply src/frontend/apps/desk/src/features/teams/components/PanelTeams.tsx]: https://github.com/numerique-gouv/people/pull/31/commits/ccef7d82b26ff1a60c90314be58c89d48aa38580

File: src/frontend/apps/desk/src/features/teams/components/PanelTeams.tsx (hunk 15 above)
Comment (lebaudantoine) [reply src/frontend/apps/desk/src/features/teams/components/PanelTeams.tsx]: Never mind, keep your implementation if it saves time, nitpicking

File: src/frontend/apps/desk/src/api/fetchApi.ts [hunk 28]
```diff
@@ -1,4 +1,4 @@
-import { useAuthStore } from '@/features/';
+import { useAuthStore } from '@/features/auth';
```
[the remark this answers is not in the document]
Comment (lebaudantoine) [reply src/frontend/apps/desk/src/api/fetchApi.ts:1]: Alright, okay, I buy it. Then the root problem comes importing multiple modules in `/features/index.ts` that create circular dependencies no? If import from `import ... from '@/features/` is gonna never be used, let's remove it, to avoid create confusion

[the remark this answers is not in the document]
Comment (lebaudantoine) [reply src/frontend/apps/desk/src/api/fetchApi.ts:1]: please feel free to ignore my comment if it makes no sense

File: src/frontend/apps/desk/src/app/globals.css (hunk 19 above)
[the remark this answers is not in the document]
Comment (lebaudantoine) [reply src/frontend/apps/desk/src/app/globals.css:25]: I am just saying, for a developer reading the code, both quantities are linked, because the actual concept is that you want the thumb to be fully rounded, based on scrollbar width. It's not explicit reading the code. And I agree it's totally okay to not couple these values that may evolve over time. Ignore my comment

File: src/frontend/apps/desk/src/features/teams/store/useTeamsStore.tsx (hunk 2 above)
Comment (lebaudantoine) [reply src/frontend/apps/desk/src/features/teams/store/useTeamsStore.tsx:19]: alright agree for the name ordering, please take a look at my first comment !

File: src/frontend/apps/desk/src/features/teams/api/useTeams.tsx (hunk 5 above)
Comment (lebaudantoine) [reply src/frontend/apps/desk/src/features/teams/api/useTeams.tsx:55]: Let's ignore my comment, we would just refactor this when switching to axios

File: src/frontend/apps/desk/src/features/teams/api/useTeams.tsx (hunk 6 above)
Comment (lebaudantoine) [reply src/frontend/apps/desk/src/features/teams/api/useTeams.tsx:27]: is it a bad practice to just ignore it in the frontend? Sorry if my question is dumb

File: docker/auth/realm.json (hunk 9 above)
Comment (lebaudantoine) [reply docker/auth/realm.json:86]: or just to add the name of the browser while creating the team ? to ensure team's unicity?

File: src/frontend/packages/i18n/__tests__/translations.test.ts (hunk 11 above)
Comment (lebaudantoine) [reply src/frontend/packages/i18n/__tests__/translations.test.ts]: Okay for the quick and dirty approach

File: src/frontend/apps/desk/src/features/teams/api/useTeams.tsx (hunk 12 above)
Comment (lebaudantoine) [reply src/frontend/apps/desk/src/features/teams/api/useTeams.tsx]: alright, good !

File: src/frontend/apps/desk/src/components/InfiniteScroll.tsx (hunk 14 above)
Comment (lebaudantoine) [reply src/frontend/apps/desk/src/components/InfiniteScroll.tsx:45]: Let's discuss it orally after the stand-up. Otherwise seems okay !

File: docker/auth/realm.json (hunk 9 above)
Comment (AntoLC) [reply docker/auth/realm.json:86]: I do it already, but it is still prone to flakiness.
Lets see:  
- chromium create 40 entries.
- my firefox tests start for any reason only after these 40 chromium entries 
- firefox tests the order, it creates 3 entries
- my 3 entries are displayed - we are ok
- firefox test click on the button sort
- The 20 oldest entries are displayed - they are the chromium ones
- firefox cannot assert that his 3 entries are sort correctly

File: src/frontend/apps/desk/src/features/teams/api/useTeams.tsx (hunk 6 above)
Comment (AntoLC) [reply src/frontend/apps/desk/src/features/teams/api/useTeams.tsx]: It is not dumb, of course it is cleaner if the back is synch with the front, but I guess it depend the case as well.
If you have a big endpoint used by different resources, with lot of useless data, to limit what your front need in the TS type make sense.

beb5da628fb16852d5a051442f3fcae0cf40f2e1

Let's remove it from the back if not needed (I will open an issue).

File: src/frontend/apps/desk/src/components/InfiniteScroll.tsx [hunk 29]
```diff
@@ -0,0 +1,55 @@
...
+      return;
+    }
+
+    const nextHandle = () => {
+      if (!hasMore || isLoading) {
+        return;
+      }
+
+      // To not wait until the end of the scroll to load more data
+      const loadBefore = 150;
```
[the remark this answers is not in the document]
Comment (AntoLC) [reply src/frontend/apps/desk/src/components/InfiniteScroll.tsx]: beb5da628fb16852d5a051442f3fcae0cf40f2e1

**Review by lebaudantoine — APPROVED** [review]
LGTM, ship it 🚢 I haven't tested locally your code after your fixup.
